changeset/changeset.go

```diff
@@ -26,18 +26,43 @@ type Config struct {
 	Decoder
 }
 
+// Valuer provides a way to get the value for a given key.
```

Comment:
WDYT about (say) `type Any interface {` that is the union of all these interfaces. Then it'd be `changeset.Any` in all external code that needs to accept any Changeset[T]. I don't think we need these fine-grained interfaces for anything, correct?